include/dylib.hpp

```diff
@@ -139,6 +143,17 @@ class dylib {
 
     explicit dylib(const char *name, bool decorations = add_filename_decorations)
         : dylib("", name, decorations) {}
+
+#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
```

Comment:
You may want to perform this check just once rather than everywhere, set a `DYLIB_CPP17` macro, then check for that.

Reply:
Done in [40e748b](https://github.com/martin-olivier/dylib/pull/53/commits/40e748be561149c59ffaf5aa812e82a3aab79b8a)